[DOCS]: GitHub CLI authentication is not used with legacy_client = false

### Expected Behavior

When no `token` or `app_auth` is configured and the user is authenticated with GitHub CLI, setting `legacy_client = false` should get a token via `gh auth token --hostname github.com` and send authenticated REST API requests.(see https://registry.terraform.io/providers/integrations/github/latest/docs#github-cli-authentication)


### Actual Behavior

With provider v6.13.0 and `legacy_client = false`, REST API requests are sent without authentication when relying on GitHub CLI authentication, and requests use the anonymous rate-limit bucket (`X-RateLimit-Limit: 60`) and fail with HTTP 403.

The debug log contains neither `Using the token from GitHub CLI.` nor `Error getting token from GitHub CLI; ensure GitHub CLI is installed and authenticated...`.

With `legacy_client = true` option, "Using the token from GitHub CLI." log is emitted and requests are authenticated.

`gh auth status` confirms that the active GitHub.com account is authenticated and stores an OAuth token in the keyring.

```
gh auth status
github.com
  ✓ Logged in to github.com account bateleurX (keyring)
  - Active account: true
  - Git operations protocol: https
  - Token: gho_************************************
  - Token scopes: ***
```

Running `gh api rate_limit` immediately after the failure shows an authenticated core limit of 5,000 with thousands of requests remaining. Explicitly setting `GITHUB_TOKEN` to the value returned by `gh auth token --hostname github.com` makes this anonymous rate-limit error disappear.

The same configuration does not produce this rate-limit error with `legacy_client = true`.

### Terraform Version

Terraform v1.15.8
on darwin_arm64
+ provider registry.terraform.io/integrations/github v6.13.0

### GitHub Installation Type

- [ ] GitHub.com (Free, Pro, or Team)
- [ ] GitHub Enterprise Server (on-premises)
- [x] GitHub Enterprise Cloud with Personal Accounts (github.com)
- [ ] GitHub Enterprise Cloud with Managed Users/EMU (github.com)
- [ ] GitHub Enterprise Cloud with Data Residency (*.ghe.com)
- [ ] I don't know

### Affected Resource(s)

- Provider authentication/configuration
- github_repository

### Terraform Configuration Files

```hcl
terraform {
  required_version = "1.15.8"
  backend "s3" {
    bucket                 = <S3 bucket name>
    key                    = <S3 key name>
    region                 = "ap-northeast-1"
    encrypt                = true
    use_lockfile           = true
    use_dualstack_endpoint = true
  }

  required_providers {
    github = {
      source  = "integrations/github"
      version = "6.13.0"
    }
  }
}

provider "github" {
  owner = <our organization>
  legacy_client = false
  parallel_requests = true
}
```

### Steps to Reproduce

1. Authenticate GitHub CLI for GitHub.com with `gh auth login`.
2. Confirm authentication with `gh auth status` and `gh auth token --hostname github.com`.
3. Ensure `GITHUB_TOKEN`, `GITHUB_APP_*`, and provider `token`/`app_auth` are not configured.
4. Configure the provider with `legacy_client = false` and an organization owner.
5. Run `TF_LOG=DEBUG terraform plan` against a configuration that reads enough REST resources to exceed the anonymous 60-request limit.
6. Observe an HTTP 403 response with `X-RateLimit-Limit: 60` and `X-RateLimit-Remaining: 0`.
7. Set `GITHUB_TOKEN="$(gh auth token --hostname github.com)"` and run the same plan again.
8. Observe that the anonymous rate-limit error no longer occurs.
9. Alternatively, set `legacy_client = true` and observe that the original error does not occur.

### Debug Output

```shell
2026-07-14T18:55:54.828+0900 [DEBUG] provider.terraform-provider-github_v6.13.0: Received HTTP Response: tf_http_op_type=response @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.40.1/helper/logging/logging_http_transport.go:162 Access-Control-Expose-Headers="ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-RateLimit-Used, X-RateLimit-Resource, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset" Content-Security-Policy="default-src 'none'; style-src 'unsafe-inline'" X-Content-Type-Options=nosniff X-Ratelimit-Resource=core Server=Varnish X-Github-Media-Type="github.v3; format=json" tf_provider_addr=registry.terraform.io/integrations/github X-Ratelimit-Used=60 tf_http_trans_id=XXXXX tf_rpc=Configure Referrer-Policy="origin-when-cross-origin, strict-origin-when-cross-origin" Strict-Transport-Security="max-age=31536000; includeSubdomains; preload" X-Ratelimit-Remaining=0
  tf_http_res_body=
  | {"message":"API rate limit exceeded for 192.0.2.10. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
   tf_http_res_version=HTTP/2.0 Content-Length=280 X-Frame-Options=deny X-Ratelimit-Limit=60 tf_http_res_status_code=403 Access-Control-Allow-Origin="*" Date="Tue, 14 Jul 2026 09:55:54 GMT" X-Xss-Protection="1; mode=block" tf_http_res_status_reason="403 Forbidden" tf_req_id=XXXXX @module=github X-Github-Request-Id=XXXXX X-Ratelimit-Reset=1784026420 Content-Type="application/json; charset=utf-8" timestamp="2026-07-14T18:55:54.828+0900"
2026-07-14T18:55:54.828+0900 [WARN]  provider.terraform-provider-github_v6.13.0: GitHub API primary rate limit detected.: @module=github category=core tf_provider_addr=registry.terraform.io/integrations/github @caller=github.com/integrations/terraform-provider-github/v6/internal/ghclient/transport.go:75 reset_time="2026-07-14T19:53:40+09:00" tf_req_id=XXXXX tf_rpc=Configure timestamp="2026-07-14T18:55:54.828+0900"
2026-07-14T18:55:54.829+0900 [ERROR] provider.terraform-provider-github_v6.13.0: Response contains error diagnostic: @module=sdk.proto tf_provider_addr=registry.terraform.io/integrations/github tf_req_id=XXXXX tf_rpc=Configure @caller=github.com/hashicorp/terraform-plugin-go@v0.31.0/tfprotov5/internal/diag/diagnostics.go:58 diagnostic_detail="" diagnostic_severity=ERROR diagnostic_summary="failed to lookup organization \"XXXXX\": Get \"https://api.github.com/orgs/XXXXX\": (http.RoundTripper).RoundTrip failed: primary rate limit reached on request to https://api.github.com/orgs/XXXXX with category: core. wait until 2026-07-14 19:53:40 +0900 JST before sending more requests." tf_proto_version=5.11 timestamp="2026-07-14T18:55:54.829+0900"
2026-07-14T18:55:54.829+0900 [ERROR] vertex "provider[\"registry.terraform.io/integrations/github\"]" error: failed to lookup organization "XXXXX": Get "https://api.github.com/orgs/XXXXX": (http.RoundTripper).RoundTrip failed: primary rate limit reached on request to https://api.github.com/orgs/XXXXX with category: core. wait until 2026-07-14 19:53:40 +0900 JST before sending more requests.
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/docs/index.md
+++ b/docs/index.md
@@ -99,6 +99,8 @@ provider "github" {
 
 ### GitHub CLI Authentication
 
+~> The GitHub CLI authentication fallback is only available when using the legacy client implementation and will be removed in a future major release. To use the legacy client implementation, set `legacy_client = true` in the provider configuration or set the `GITHUB_LEGACY_CLIENT` environment variable to `true`.
+
 When using the GitHub CLI authentication fallback, you can optionally specify the path to the `gh` executable using the `GH_PATH` environment variable. This is useful when the provider cannot properly determine the path to GitHub CLI, such as in cygwin terminals. If not specified, the provider looks for `gh` in your system PATH.
 
 #### .env

--- a/templates/index.md.tmpl
+++ b/templates/index.md.tmpl
@@ -63,6 +63,8 @@ Some API operations may not be available when using a GitHub App installation co
 
 ### GitHub CLI Authentication
 
+~> The GitHub CLI authentication fallback is only available when using the legacy client implementation and will be removed in a future major release. To use the legacy client implementation, set `legacy_client = true` in the provider configuration or set the `GITHUB_LEGACY_CLIENT` environment variable to `true`.
+
 When using the GitHub CLI authentication fallback, you can optionally specify the path to the `gh` executable using the `GH_PATH` environment variable. This is useful when the provider cannot properly determine the path to GitHub CLI, such as in cygwin terminals. If not specified, the provider looks for `gh` in your system PATH.
 
 #### .env

